core/src/main/kotlin/fr/sncf/osrd/api/api_v2/RequirementsParser.kt

```diff
@@ -66,3 +79,89 @@ fun parseRoutingRequirements(
     }
     return res
 }
+
+fun parseWorkSchedulesRequest(
+    infra: RawSignalingInfra,
+    workSchedulesRequest: WorkSchedulesRequest,
+    startTime: ZonedDateTime
+): Collection<Requirements> {
+    val delta = TimeDelta(between(startTime, workSchedulesRequest.startTime).toMillis())
+    return convertWorkSchedules(infra, workSchedulesRequest.workScheduleRequirements, delta)
+}
+
+/**
+ * Convert work schedules into timetable spacing requirements. This is not entirely semantically
+ * correct, but it lets us avoid work schedules like any other kind of time-bound constraint
+ */
+fun convertWorkSchedules(
+    rawInfra: RawSignalingInfra,
+    workSchedules: Map<Long, WorkSchedule>,
+    timeToAdd: TimeDelta = 0.seconds
+): Collection<Requirements> {
+    val res = mutableListOf<Requirements>()
+    for (entry in workSchedules) {
+        val workScheduleRequirements = mutableListOf<ResultTrain.SpacingRequirement>()
+        workScheduleRequirements.addAll(convertWorkSchedule(rawInfra, entry.value, timeToAdd))
+        res.add(
+            Requirements(
+                RequirementId(entry.key, RequirementType.WORK_SCHEDULE),
+                workScheduleRequirements,
+                listOf()
+            )
+        )
+    }
+    return res
+}
+
+/**
+ * Convert work schedules into timetable spacing requirements, without taking work schedule id into
+ * account. This is not entirely semantically correct, but it lets us avoid work schedules like any
+ * other kind of time-bound constraint
+ */
+fun convertWorkSchedules(
```

Comment:
Is it intended to have a function with the exact same name than the one above, but with different input arguments? I know I'm not used to overloading (not available in the languages I deal with), but yet, I find this extremely misleading since both function don't even return the same type.

Reply:
It was indeed intended. They technically perform the same operation, albeit for different structures : one is a collection of work_chedules, so it only outputs one Requirements, the other is a map of work_schedules, hence it outputs a collection of Requirements. Which is why I kept the same name for both. It also is used for the same reason : one converts the work_schedules in stdcm, the other in conflict_detection. Honestly think it's fine, especially since I don't really think there's a simple naming alternative. Don't hesitate to challenge this if you don't find the explanation satisfying.

Reply:
There are okay alternatives. Especially since it's confusing for 2 people already, I'll definitely do the change.